### PR TITLE
Widgets: Add Welcome Guide

### PR DESCRIPTION
**Pull request by noisysocks**
## Description

Requires https://github.com/WordPress/gutenberg/pull/31926.
Implements part of https://github.com/WordPress/gutenberg/issues/30887.

Adds `WelcomeGuide` to the widgets screen. This is mostly copied from `WelcomeGuide` in `edit-post`.

We show the "new user" copy when all widgets are block widgets, because when this feature is merged to Core, we will update `wp_install_defaults` ([here](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/includes/upgrade.php#L404)) to set up a sidebar containing block widgets in new WordPress installations. We can't do this in the Gutenberg plugin, though.

## How has this been tested?

1. Go to Appearance → Widgets. You should see the welcome guide.
2. Dismiss the guide and refresh the page. You should not see the welcome guide.
3. Replace every Legacy Widget with a block.
4. Clear local storage e.g. by running `localStorage.clear()`.
5. Refresh the page. You should see the welcome guide with different copy.

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/edit-widgets/src/components/welcome-guide/index.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	EditorImage,
+	BlockLibraryImage,
+	DocumentationImage,
+	InserterIconImage,
+} from './images';
+import { store as editWidgetsStore } from '../../store';
+
+export default function WelcomeGuide() {
+	// const isActive = useSelect(
+	// 	( select ) => select( editWidgetsStore ).isFeatureActive( 'welcomeGuide' ),
```
Comment (noisysocks) [inline packages/edit-widgets/src/components/welcome-guide/index.js]: I think @talldan is adding this so I'll wait for him 🙂

**Review by talldan — APPROVED** [review]
I've rebased to make the diff cleaner for review.

Works well and code looks good, nice work!

**Comment by noisysocks** [comment]
> +277 kB (+1918%)

This really highlights how `images.js` should maybe belong in a shared package, or maybe in `wp-admin/images/`. I'll add a note to https://github.com/WordPress/gutenberg/issues/31965.

**Comment by ellatrix** [comment]
Why was this merged with failing tests? It's not because trunk is failing that that's ok?

**Comment by paaljoachim** [comment]
Testing.

WordPress 5.7.2. Theme: Twenty Twenty.
Special Gutenberg PR version.

This is what I see:

https://user-images.githubusercontent.com/5323259/118809158-8af20500-b8aa-11eb-8afc-027e8d6e261a.mp4

Thank you for adding this feature!

**Pull request merged**